src/librustc/hir/map/mod.rs

```diff
@@ -168,43 +168,48 @@ impl<'hir> MapEntry<'hir> {
         })
     }
 
-    fn is_body_owner(self, node_id: NodeId) -> bool {
+    fn associated_body(self) -> Option<BodyId> {
```

Comment:
I have at least two, if not three, reimplementations of this, seems so obvious in retrospect!